### PR TITLE
Basic web radio support: fill artist from title (#5)

### PR DESCRIPTION
**Pull request by xabolcs**
This is my ~quick and~ dirty hack of basic web radio support (#5).

Examples

Screenshot from 2021-07-02 11-57-35

Screenshot from 2021-07-02 11-59-18

### CODE REVIEW

**Comment by xabolcs** [comment]
Sure, I had to do my [homework](https://www.freedesktop.org/wiki/Specifications/mpris-spec/metadata/#xesam:artist), but I just don't understand why it worked first and why it doesn't work now. :thinking: 
```
gnome-shell[3620]: Received faulty track artist metadata from org.mpris.MediaPlayer2.xmms2; expected an array of strings, got Green Day (string)
```

----

It does work, "just" pollutes the log with that message. 🙄

**Comment by w0rp** [comment]
I've been away for a while, but I'll read this code carefully later and try to merge it in.

There's another issue at the moment where the media keys don't work any more in KDE, and I want to fix that.

File: src/xmms2-mpris.c [hunk 1]
```diff
@@ -190,11 +190,26 @@ void handle_track_info(XmmsTrackInfo* info) {
...
+    if (!(info->artist && info->artist[0] != '\0')) {
+        webtitle_array = g_strsplit(info->title, " - ", 2);
+        if (g_strv_length(webtitle_array) < 2) {
+            webtitle_array = g_strsplit(info->title, " / ", 2);
+        }
+        if (g_strv_length(webtitle_array) >= 2) {
+            info->artist = webtitle_array[0];
+            info->title = webtitle_array[1];
+        }
+    }
```
Comment (w0rp) [inline src/xmms2-mpris.c]:
```suggestion

    if (!(info->artist && info->artist[0] != '\0')) {
        webtitle_array = g_strsplit(info->title, " - ", 2);

        if (g_strv_length(webtitle_array) < 2) {
            webtitle_array = g_strsplit(info->title, " / ", 2);
        }

        if (g_strv_length(webtitle_array) >= 2) {
            info->artist = webtitle_array[0];
            info->title = webtitle_array[1];
        }
    }
```

This is good. The only change I ask for is just a bit more whitespace around blocks, which I'll add now.

**Review by w0rp — APPROVED** [review]

**Comment by w0rp** [comment]
Now in `master`. Thanks!

**Comment by xabolcs** [comment]
> ```gnome-shell[3620]: Received faulty track artist metadata from org.mpris.MediaPlayer2.xmms2; expected an array of strings, got Green Day (string)```

And what about this pollution? :roll_eyes: 
Should I open another issue about it?

The problem is that this plugin sends the data in wrong structure:

> `xesam:artist`
>
> List of Strings: The track artist(s).

And the [current code](https://github.com/w0rp/xmms2-mpris/blob/d647b687cfdbea9ddd9f3100c059ae6c501e56c4/src/mpris.c#L47-L49) uses `new_metadata_dict_string`
```
    if (info->artist && info->artist[0] != '\0') {
        entries[length++] = new_metadata_dict_string("xesam:artist", info->artist);
    }
```

**Comment by w0rp** [comment]
If it still doesn't quite work properly, I'll reopen the issue.

**Pull request merged**